cns/restserver/api.go

```diff
@@ -899,6 +899,95 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 	logger.Response(service.Name, getNetworkContainerResponse, returnCode, err)
 }
 
+// getAllNetworkContainers is used to check if NC refresh association is needed
+// If any NC from DNC request is not existing in CNS, refresh association will be triggered
+func (service *HTTPRestService) getAllNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getAllNetworkContainers")
+	if r.Method != "GET" {
+		logger.Printf("[Azure CNS] Error. getAllNetworkContainers did not receive a GET.")
+		return
+	}
+
+	var req cns.GetAllNetworkContainersRequest
+	err := service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		logger.Printf("[Azure CNS] getAllNetworkContainers failed with error: %s", err.Error())
+		return
+	}
+
+	ncDetails, allNcExist := service.getAllNetworkContainersDetails(req.NetworkContainersID)
+
+	response := cns.GetAllNetworkContainersResponse{
+		ContainerStatus:           ncDetails,
+		AllNetworkContainersExist: allNcExist,
+		Response: cns.Response{
+			ReturnCode: types.Success,
+		},
+	}
+
+	err = service.Listener.Encode(w, &response)
+	logger.Response(service.Name, response, response.Response.ReturnCode, err)
+}
+
+// refreshNetworkContainers receives and stores the GS for all the NCs of node from DNC
+func (service *HTTPRestService) refreshNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] refreshNetworkContainers")
+	if r.Method != "POST" {
+		logger.Printf("[Azure CNS] Error. refreshNetworkContainers did not receive a POST.")
+		return
+	}
+
+	var req cns.CreateNetworkContainerRequest
```

Comment:
This is information for a single NC. We need a new type that allows the client to send multiple NCs.

Reply:
Addressed.